server/routerlicious/packages/lambdas/src/deli/lambda.ts

```diff
@@ -963,14 +1084,61 @@ export class DeliLambda extends TypedEventEmitter<IDeliLambdaEvents> implements
             timestamp: Date.now(),
             type: NackOperationType,
         };
+
         return {
-            instruction: InstructionType.NoOp,
+            ticketType: TicketType.Nack,
             message: nackMessage,
-            msn: this.minimumSequenceNumber,
-            nacked: true,
-            send: SendType.Immediate,
-            timestamp: message.timestamp,
-            type: message.operation.type,
+        };
+    }
+
+    /**
+     * Creates a signal message for clients.
+     */
+    private createSignalMessage(
+        message: IRawOperationMessage,
+        sequenceNumber: number,
+        dataContent: any): ISignalMessageOutput {
+        let signalMessage: ISignalMessage;
+
+        switch (message.operation.type) {
+            case MessageType.ClientJoin:
+                signalMessage = createRoomJoinMessage(
+                    (dataContent as IClientJoin).clientId,
+                    (dataContent as IClientJoin).detail);
+                break;
+
+            case MessageType.ClientLeave:
+                signalMessage = createRoomLeaveMessage(
+                    typeof (dataContent) === "string" ? dataContent : dataContent.clientId);
+                break;
+
+            case MessageType.Control:
+                // this will tell broadcaster to process the control message the client
+                signalMessage = {
+                    clientId: null,
+                    content: JSON.stringify({
+                        type: MessageType.Control,
+                        content: dataContent,
+                    }),
+                };
+                break;
+
+            default:
+                throw new Error(`Cannot create signal message for type ${message.operation.type}`);
```

Comment:
you can use unreachableCase(), it's a bit stronger as it uses compiler support for detecting reachable paths

Reply:
`type` is just a string so it gives me this error
![image](https://user-images.githubusercontent.com/41303831/155225175-a1ddce64-7182-4f4d-8214-254d82f551c1.png)